test/dynamo/test_repros.py

```diff
@@ -3722,6 +3722,7 @@ def fn(input, mask):
         actual = fn_opt(*inputs)
         self.assertTrue(same(actual, expected))
 
+    @unittest.skip("Temparory for testing")
```

Comment:
Issue 1 at https://github.com/pytorch/pytorch/pull/142830#issuecomment-2552763164

Reply:
Do you know where the issue is coming from? I think the problem is that torch.compile really wants to know if the Tensor is contiguous or not. Unfortunately, it cannot tell this if the tensor has all unbacked symints. If the user code is calling is_contiguous() on the Tensor -- then this error makes sense and we should figure out how to raise a better error. If torch.compile is calling is_contiguous() on the tensor, then we should try to figure out how to get it to not do it.

Reply:
This is a good point! Let me explain the root cause use the failed unit test [`test_save_for_bwd`](https://github.com/pytorch/pytorch/blob/75661f2036d36fd7f869cd749eb6ef5fb40e4772/test/dynamo/test_autograd_function.py#L324). The error originates from the fake propagation during the tracing of its backward. Specifically, I believe the fake propagation of the elementwise sqrt function triggers the [`guard_size_oblivious`](https://github.com/pytorch/pytorch/blob/75661f2036d36fd7f869cd749eb6ef5fb40e4772/torch/_prims_common/__init__.py#L517), which subsequently causes a dynamic shape error: ```Could not guard on data-dependent expression u1 < u0 (unhinted: u1 < u0)```.
Why didn’t this error occur previously? It’s because `sqrt` was not considered a banned operation in strict mode before. In the past, we simply used `fwd_out` as `grad_out` to trace the backward function, bypassing the issue.
I also reviewed another failed unit test: `test_hf_xsoftmax_training`. It has the same root cause, but the issue is triggered by a different meta function.

Reply:
It seems this is either a bug or a limitation in the meta functions with dynamic shapes. If a tensor’s stride is represented by an unbacked SymInt, many operations relying on fake propagation may not function correctly.